app/controllers/shipments_controller.rb

```diff
@@ -1,6 +1,7 @@
 class ShipmentsController < ApplicationController
   def index
-    @shipments = Shipment.all
+    company_id = Company.custom_find_id(params[:company_id])
+    @shipments = Shipment.all_items(company_id)
```

Comment:
You might be overthinking, my personal idea for the task here is mainly:
1. You already have association `shipments` under company, so it's supposed to be `company.shipments`
2. shipment_items called in jbuilder, so `includes` for `company.shipments.includes(:shipment_items) to avoid n+1 query